From 97a9aab8dfab5ed65a03c5629d94a98042f419b3 Mon Sep 17 00:00:00 2001
From: Johnanater <10360610+Johnanater@users.noreply.github.com>
Date: Thu, 22 Aug 2019 20:33:36 -0400
Subject: [PATCH] Restore online-mode functionality

You will need a client patch in order to use online-mode. Thanks to
Johnanater for the fixed URL!

diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 28178ba9..e789839f 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -179,12 +179,17 @@ public class MinecraftServer implements Runnable, ICommandListener {
             return false;
         }
 
+        // Canyon start - note on online/offline mode
         if (!this.onlineMode) {
             log.warning("**** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!");
             log.warning("The server will make no attempt to authenticate usernames. Beware.");
             log.warning("While this makes the game possible to play without internet access, it also opens up the ability for hackers to connect with any username they choose.");
-            log.warning("To change this, set \"online-mode\" to \"true\" in the server.settings file.");
+            log.warning("To change this, set \"online-mode\" to \"true\" in the server.settings file. Note that a client mod is required for this to work.");
+        } else {
+            log.warning("**** SERVER IS RUNNING IN ONLINE MODE!");
+            log.warning("Clients who wish to connect to this server must be patched to use the correct session endpoint.");
         }
+        // Canyon end
 
         this.serverConfigurationManager = new ServerConfigurationManager(this);
         // CraftBukkit - removed trackers
diff --git a/src/main/java/net/minecraft/server/ThreadLoginVerifier.java b/src/main/java/net/minecraft/server/ThreadLoginVerifier.java
index 4d98e6ad..d9ff03f4 100644
--- a/src/main/java/net/minecraft/server/ThreadLoginVerifier.java
+++ b/src/main/java/net/minecraft/server/ThreadLoginVerifier.java
@@ -30,7 +30,7 @@ class ThreadLoginVerifier extends Thread {
     public void run() {
         try {
             String s = NetLoginHandler.a(this.netLoginHandler);
-            URL url = new URL("http://www.minecraft.net/game/checkserver.jsp?user=" + URLEncoder.encode(this.loginPacket.name, "UTF-8") + "&serverId=" + URLEncoder.encode(s, "UTF-8"));
+            URL url = new URL("http://session.minecraft.net/game/checkserver.jsp?user=" + URLEncoder.encode(this.loginPacket.name, "UTF-8") + "&serverId=" + URLEncoder.encode(s, "UTF-8")); // Canyon - Fix online mode for server
             BufferedReader bufferedreader = new BufferedReader(new InputStreamReader(url.openStream()));
             String s1 = bufferedreader.readLine();
 
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 4f1902fc..810778b9 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -365,7 +365,7 @@ public final class CraftServer implements Server {
         boolean animals = config.getBoolean("spawn-animals", console.spawnAnimals);
         boolean monsters = config.getBoolean("spawn-monsters", console.worlds.get(0).spawnMonsters > 0);
 
-        console.onlineMode = config.getBoolean("online-mode", console.onlineMode);
+        //console.onlineMode = config.getBoolean("online-mode", console.onlineMode); // Canyon - online mode doesn't work.
         console.spawnAnimals = config.getBoolean("spawn-animals", console.spawnAnimals);
         console.pvpMode = config.getBoolean("pvp", console.pvpMode);
         console.allowFlight = config.getBoolean("allow-flight", console.allowFlight);
-- 
2.24.1.windows.2

